File: fastapi/fastapi-data/common/utils/response.py
from typing import Any, Optional, Dict

def success(data: Optional[Any] = None, msg: str = "success", code: int = 1) -> Dict[str, Any]:
    return {
        "code": code,
        "data": data,
        "msg": msg
    }

def fail(msg: str = "error", code: int = 0, data: Optional[Any] = None) -> Dict[str, Any]:
    return {
        "code": code,
        "data": data,
        "msg": msg
    }

File: gin/common/utils/response.go
package utils

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

// Success 返回成功响应
func Success(c *gin.Context, data any) {
	c.JSON(http.StatusOK, gin.H{
		"code": 1,
		"msg":  "success",
		"data": data,
	})
	c.Abort()
}

// Error 返回错误响应
func Error(c *gin.Context, code int, message string) {
	c.JSON(code, gin.H{
		"code": 0,
		"msg":  message,
		"data": nil,
	})
	c.Abort()
}
